packages/database/src/migrations/20201016001619-RemoveMugilFromStrive-modifies-data.js

```diff
@@ -0,0 +1,163 @@
+'use strict';
+
+import { arrayToDbString, generateId, insertObject } from '../utilities';
+
+var dbm;
+var type;
+var seed;
+
+/**
+  * We receive the dbmigrate dependency from dbmigrate initially.
+  * This enables us to not have to rely on NODE_PATH.
+  */
+exports.setup = function(options, seedLink) {
+  dbm = options.dbmigrate;
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+const getHierarchyId = async (db) => {
+  const results = await db.runSql(`SELECT id FROM entity_hierarchy WHERE name = 'strive';`);
+  if (results.rows.length > 0) {
+    return results.rows[0].id;
+  }
+  throw new Error(`Strive hierarchy not found`);
+}
+
+const getEntityIdByCode = async (db, code) => {
+  const results = await db.runSql(`SELECT id FROM entity WHERE code = '${code}';`);
+  if (results.rows.length > 0) {
+    return results.rows[0].id;
+  }
+  throw new Error(`Entity with code ${code} Not found`);
+}
+
+const getEntityIdsByParentId = async (db, parentId) => {
+  const results = await db.runSql(`SELECT id FROM entity WHERE parent_id = '${parentId}';`);
+  return results.rows.map(row => row.id);
+}
+
+exports.up = async function(db) {
+  const hierarchyId = await getHierarchyId(db);
+
+  // 1. Create alternative project hierarchy without Mugil
+  const entityIds = {
+    PG: await getEntityIdByCode(db, 'PG'),             // country
+    Madang: await getEntityIdByCode(db, 'PG_Madang'),  // district
+    Mugil: await getEntityIdByCode(db, 'PG_Mugil'),    // facility
+    Sausi: await getEntityIdByCode(db, 'PG_MADANG01'), // facility
+  };
+
+  // 1.1. inserts all districts
+  const districtIds = await getEntityIdsByParentId(db, entityIds.PG);
+
+  for (const districtId of districtIds) {
+    await insertObject(db, 'entity_relation', {
+      id: generateId(),
+      parent_id: entityIds.PG,
+      child_id: districtId,
+      entity_hierarchy_id: hierarchyId
+    });
+  }
+
+  // 1.2. generate list of facilities to insert
+  const facilityIdsToInsert = [];
+  for (const districtId of districtIds) {
+    const facilityIds = await getEntityIdsByParentId(db, districtId);
+    for (const facilityId of facilityIds) {
+      if (facilityId === entityIds.Mugil) {
+        continue;
+      }
+      facilityIdsToInsert.push(facilityId);
+    }
+  }
+  facilityIdsToInsert.push(entityIds.Sausi);
+
+  // 1.3. insert facilities
+  await db.runSql(`
+    INSERT INTO entity_relation (id, child_id, parent_id, entity_hierarchy_id)
+    SELECT
+      generate_object_id() as id, 
+      entity.id as child_id, 
+      entity.parent_id as parent_id, 
+      '${hierarchyId}' as entity_hierarchy_id 
+    FROM entity 
+    WHERE id IN (${arrayToDbString(facilityIdsToInsert)});
+  `);
+
+  // 1.3. insert villages for each facility inserted
+  await db.runSql(`
```

Comment:
I haven't done any development work in entity hierarchies but I've done reviewing, and my impression is that we just need to define entity relations for the level we want to customise: in this case, it would be defining entity relations for Strive facilities.

This means that we don't need to create entity relations for villages, since we will retrieve those via the standard hierarchy: they will be attached to their parent facilities via `parent_id` in the `entity` table. Similarly, we wouldn't need to create entity relations for districts because no customisation is needed for them. @edmofro could you please confirm this?

Reply:
I double checked and you are completely right.

Not sure why it didn't work for me like that the first time, but changing to just insert facility level (without Mugil).